pkg/kubectl/cmd/util/openapi/validation/validation.go

```diff
@@ -76,14 +76,14 @@ func (v *SchemaValidation) validateList(object interface{}) []error {
 }
 
 func (v *SchemaValidation) validateResource(obj interface{}, gvk schema.GroupVersionKind) []error {
-	if !api.Registry.IsEnabledVersion(gvk.GroupVersion()) {
-		// if we don't have this in our scheme, just skip
-		// validation because its an object we don't recognize
-		return nil
-	}
-
 	resource := v.resources.LookupResource(gvk)
 	if resource == nil {
+		if !api.Registry.IsEnabledVersion(gvk.GroupVersion()) {
+			// if we don't have this in our scheme, just skip
+			// validation because its an object we don't recognize
+			return nil
+		}
+
 		return []error{fmt.Errorf("unknown object type %#v", gvk)}
```

Comment:
I'm not sure this should be a different case anymore.  If we don't have the information to validate, we can't validate.  I don't know that having a matching type in the scheme ought to matter either way.

Reply:
The openapi swagger has everything it needs to validate the payload. And we want to validate custom-types, like service-catalog resources, which are never going to be hard-coded in kubectl.

Reply:
@deads2k Can I has lgtm?

Reply:
> The openapi swagger has everything it needs to validate the payload. And we want to validate custom-types, like service-catalog resources, which are never going to be hard-coded in kubectl.

I think my comment was unclear.  If we can't find the resource, why don't we always return nil?

Reply:
Yeah, we just discussed with @pwittrock. Let's get rid of that condition